### ISSUE TITLE
gstreamer for steam-run build broken on master

### ISSUE THREAD

**Issue by marzipankaiser** [NixOS/nixpkgs#63635]
## Issue description
Build fails for `steam-run` when using current nixpkgs master:
    
```
$ nix-shell '<nixpkgs>' -A steam-run --pure
these derivations will be built:
  /nix/store/vzvp6d8mml24f0f98a1c4pb6gpacd9kh-gst-plugins-base-1.16.0.drv
  /nix/store/xn3kz23bpi2i2h96b89y2fmq7qsfz4g3-gst-plugins-ugly-1.16.0.drv
  /nix/store/831qz5iw51p27fja2fq0vy7v820zkys4-steam-run-usr-multi.drv
  /nix/store/g9qgp240gmj8hgg0k771hay72qlr2p3r-gst-plugins-base-1.16.0.drv
  /nix/store/4vxlg34v3lfx8i6jzhdbjmx1zf5kpv72-gst-plugins-ugly-1.16.0.drv
  /nix/store/vz2fxafddp9zfl09jcran3x790idbi49-steam-run-usr-target.drv
  /nix/store/080al2r06g3f03wa0sxvq7c7b3riz6l7-steam-run-fhs.drv
  /nix/store/20cnnn682bki449wnx2rz3xmcj8amjpl-steam-run-init.drv
building '/nix/store/g9qgp240gmj8hgg0k771hay72qlr2p3r-gst-plugins-base-1.16.0.drv'...
building '/nix/store/vzvp6d8mml24f0f98a1c4pb6gpacd9kh-gst-plugins-base-1.16.0.drv'...
unpacking sources
unpacking source archive /nix/store/ny4rkf83c82k6vn17vpzpq6y4p8f1r3i-gst-plugins-base-1.16.0.tar.xz
unpacking sources
unpacking source archive /nix/store/ny4rkf83c82k6vn17vpzpq6y4p8f1r3i-gst-plugins-base-1.16.0.tar.xz
source root is gst-plugins-base-1.16.0
source root is gst-plugins-base-1.16.0
setting SOURCE_DATE_EPOCH to timestamp 1555629612 of file gst-plugins-base-1.16.0/po/zh_CN.po
patching sources
applying patch /nix/store/x7py1q1yqdc1kkk9k469qi24mm83j7rd-fix_pkgconfig_includedir.patch
patching file pkgconfig/meson.build
patching script interpreter paths in .
setting SOURCE_DATE_EPOCH to timestamp 1555629612 of file gst-plugins-base-1.16.0/po/zh_CN.po
patching sources
applying patch /nix/store/x7py1q1yqdc1kkk9k469qi24mm83j7rd-fix_pkgconfig_includedir.patch
patching file pkgconfig/meson.build
./depcomp: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
patching script interpreter paths in .
./config.sub: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./depcomp: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./install-sh: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./config.sub: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./test-driver: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./install-sh: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./common/extract-release-date-from-doap-file: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./test-driver: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./common/download-translations: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./common/extract-release-date-from-doap-file: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./common/download-translations: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./common/check-exports: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./common/scangobj-merge.py: interpreter directive changed from "/usr/bin/python" to "/nix/store/wd9cd2hhfim7i64ff1p1vsm6cb5kf4gq-python3-3.7.3/bin/python"
./common/check-exports: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./common/gst-indent: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./common/scangobj-merge.py: interpreter directive changed from "/usr/bin/python" to "/nix/store/4c4ajgdnhlqk994hilagk5cgv7vw9yzg-python3-3.7.3/bin/python"
./compile: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./common/gst-indent: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./config.guess: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./compile: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./autogen.sh: interpreter directive changed from "/bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./config.guess: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./config.rpath: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./autogen.sh: interpreter directive changed from "/bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./missing: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./config.rpath: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
./configure: interpreter directive changed from " /bin/sh" to "/nix/store/j1qjqdwxlyc62fadqw4kya5cn0qyf2sc-bash-4.4-p23/bin/sh"
./missing: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
configuring
./configure: interpreter directive changed from " /bin/sh" to "/nix/store/w9ngash2dw3pvl98ysd79qy2rkkmc8my-bash-4.4-p23/bin/sh"
meson flags: --buildtype=release         --libdir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/lib --libexecdir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/libexec         --bindir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/bin --sbindir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/sbin         --includedir=/nix/store/47yy0agwxzl5wks4zjjilpcgk66mawai-gst-plugins-base-1.16.0-dev/include         --mandir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/share/man --infodir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/share/info         --localedir=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0/share/locale         -Dauto_features=disabled         --prefix=/nix/store/qls86h2ipf9x79i05msba23v4rbb9wic-gst-plugins-base-1.16.0 -Dauto_features=enabled -Dexamples=disabled -Dgl-graphene=disabled -Dgl_winsys=[x11,wayland] -Dgtk_doc=disabled
configuring
meson flags: --buildtype=release         --libdir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/lib --libexecdir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/libexec         --bindir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/bin --sbindir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/sbin         --includedir=/nix/store/mgw15jqamf2jw0qqfdch3vb68qjm8b7w-gst-plugins-base-1.16.0-dev/include         --mandir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/share/man --infodir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/share/info         --localedir=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0/share/locale         -Dauto_features=disabled         --prefix=/nix/store/hzlqdrgln5kmlp3sqdl8nblhlpi5kqd0-gst-plugins-base-1.16.0 -Dauto_features=enabled -Dexamples=disabled -Dgl-graphene=disabled -Dgl_winsys=[x11,wayland] -Dgtk_doc=disabled
gcc: error: /build/gst-plugins-base-1.16.0/build/meson-private/sanitycheckobjc.m: Objective-C compiler not installed on this system
gcc: error: /build/gst-plugins-base-1.16.0/build/meson-private/sanitycheckobjc.m: Objective-C compiler not installed on this system
The Meson build system
Version: 0.49.2
Source dir: /build/gst-plugins-base-1.16.0
Build dir: /build/gst-plugins-base-1.16.0/build
Build type: native build
Project name: gst-plugins-base
Project version: 1.16.0
Native C compiler: /nix/store/x3fl1fblijxrmnsiwipsssqhpffz168r-gcc-wrapper-7.4.0/bin/cc (gcc 7.4.0 "gcc (GCC) 7.4.0")
Build machine cpu family: x86
Build machine cpu: i686
Native C++ compiler: /nix/store/x3fl1fblijxrmnsiwipsssqhpffz168r-gcc-wrapper-7.4.0/bin/c++ (gcc 7.4.0 "g++ (GCC) 7.4.0")
Compiler for language objc not found.
Compiler for C supports link arguments -Wl,-Bsymbolic-functions: YES
Compiler for C supports arguments -fvisibility=hidden: YES
Compiler for C supports arguments -fno-strict-aliasing: YES
Has header "dlfcn.h" : YES
Has header "emmintrin.h" : YES
Has header "inttypes.h" : YES
Has header "memory.h" : YES
Has header "process.h" : NO
Has header "smmintrin.h" : YES
Has header "stdint.h" : YES
Has header "stdlib.h" : YES
Has header "strings.h" : YES
Has header "string.h" : YES
Has header "sys/socket.h" : YES
Has header "sys/stat.h" : YES
Has header "sys/types.h" : YES
Has header "sys/wait.h" : YES
Has header "unistd.h" : YES
Has header "winsock2.h" : NO
Has header "xmmintrin.h" : YES
Has header "linux/dma-buf.h" : YES
Library m found: YES
Checking for function "dcgettext" with dependency -lm: YES
Checking for function "gmtime_r" with dependency -lm: YES
Checking for function "lrintf" with dependency -lm: YES
Checking for function "mmap" with dependency -lm: YES
Checking for function "log2" with dependency -lm: YES
Checking for size of "char" : 1
Checking for size of "int" : 4
Checking for size of "long" : 4
Checking for size of "short" : 2
Checking for size of "void*" : 4
Compiler for C++ supports arguments -Waggregate-return: YES
Compiler for C supports arguments -Wmissing-declarations: YES
Compiler for C++ supports arguments -Wmissing-declarations: YES
Compiler for C supports arguments -Wredundant-decls: YES
Compiler for C++ supports arguments -Wredundant-decls: YES
Compiler for C supports arguments -Wundef: YES
Compiler for C++ supports arguments -Wundef: YES
Compiler for C supports arguments -Wwrite-strings: YES
Compiler for C++ supports arguments -Wwrite-strings: YES
Compiler for C supports arguments -Wformat: YES
Compiler for C++ supports arguments -Wformat: YES
Compiler for C supports arguments -Wformat-nonliteral: YES
Compiler for C++ supports arguments -Wformat-nonliteral: YES
Compiler for C supports arguments -Wformat-security: YES
Compiler for C++ supports arguments -Wformat-security: YES
Compiler for C supports arguments -Winit-self: YES
Compiler for C++ supports arguments -Winit-self: YES
Compiler for C supports arguments -Wmissing-include-dirs: YES
Compiler for C++ supports arguments -Wmissing-include-dirs: YES
Compiler for C supports arguments -Waddress: YES
Compiler for C++ supports arguments -Waddress: YES
Compiler for C supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C++ supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C supports arguments -Wvla: YES
Compiler for C++ supports arguments -Wvla: YES
Compiler for C supports arguments -Wpointer-arith: YES
Compiler for C++ supports arguments -Wpointer-arith: YES
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Wdeclaration-after-statement: YES
Found pkg-config: /nix/store/y1anxcj45n8w4pbzjlq5m10b2fnhc3jw-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Dependency x11 found: YES 1.6.7
Dependency glib-2.0 found: YES 2.60.3
Dependency gobject-2.0 found: YES 2.60.3
Dependency gio-2.0 found: YES 2.60.3
Dependency gio-unix-2.0 found: YES 2.60.3
Dependency gmodule-no-export-2.0 found: YES 2.60.3
Dependency gdk-pixbuf-2.0 skipped: feature examples disabled
Dependency gtk+-3.0 skipped: feature examples disabled
Dependency gtk+-x11-3.0 skipped: feature examples disabled
Dependency gstreamer-1.0 found: YES 1.16.0
Dependency gstreamer-base-1.0 found: YES 1.16.0
Dependency gstreamer-net-1.0 found: YES 1.16.0
Found CMake: NO
Dependency gstreamer-check-1.0 found: NO (tried pkgconfig)
Looking for a fallback subproject for the dependency gstreamer-check-1.0
Couldn't use fallback subproject in subprojects/gstreamer for the dependency gstreamer-check-1.0
Reason: Failed to initialize 'subprojects/gstreamer':
Subproject directory not found and gstreamer.wrap file not found

meson.build:303:0: ERROR:  Dependency "gstreamer-check-1.0" not found, tried pkgconfig

A full log can be found at /build/gst-plugins-base-1.16.0/build/meson-logs/meson-log.txt
builder for '/nix/store/vzvp6d8mml24f0f98a1c4pb6gpacd9kh-gst-plugins-base-1.16.0.drv' failed with exit code 1
cannot build derivation '/nix/store/xn3kz23bpi2i2h96b89y2fmq7qsfz4g3-gst-plugins-ugly-1.16.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/831qz5iw51p27fja2fq0vy7v820zkys4-steam-run-usr-multi.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/080al2r06g3f03wa0sxvq7c7b3riz6l7-steam-run-fhs.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/20cnnn682bki449wnx2rz3xmcj8amjpl-steam-run-init.drv': 1 dependencies couldn't be built
error: build of '/nix/store/20cnnn682bki449wnx2rz3xmcj8amjpl-steam-run-init.drv' failed

```
The meson-logs mentioned in the output do not exist anywhere as far as I can tell.

The build works fine on a718f72a8e88a97bb4381a4ece5d63c69ab1a217.

### Steps to reproduce
Try to build `steam-run` from current master (987ec8aed453ea944a4d53aa0ee47d8371c6844c).

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 5.1.12, NixOS, 19.09pre-git (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `""`
 - channels(gaisseml): `""`
 - nixpkgs: `/nix/store/rs0pi5j63ihrvfalfc6q5cy02q9x4rd2-src/nixpkgs`

`nix-info` output looks that way since I build from a nixpkgs git checkout and do not use channels.

**Comment by marzipankaiser** [comment]
building `gst-plugins-base` works, since this builds a different version (0.10.36)

**Comment by vcunat** [comment]
Fixed by #63637.
